mono/utils/mono-threads-coop.c

```diff
@@ -164,13 +205,16 @@ mono_threads_reset_blocking_end (void *cookie)
 		return;
 
 	g_assert (info == mono_thread_info_current_unchecked ());
-	mono_threads_prepare_blocking ();
+	mono_threads_prepare_blocking (stackdata);
 }
 
 void*
-mono_threads_try_prepare_blocking (void)
+mono_threads_try_prepare_blocking (void* stackdata)
 {
 	MonoThreadInfo *info;
+	void* stackdata_end;
+
+	SAVE_REGS_ON_STACK;
```

Comment:
This is the wrong place to use __builtin_unwind_init.

It should be used in the function that does the stack copying (copy_stack_data), this way we ensure to have visibility into all regs.